packages/drivers/odsp-driver-definitions/src/odspCache.ts

```diff
@@ -127,5 +127,24 @@ export interface IPersistedCache {
  * @internal
  */
 export function getKeyForCacheEntry(entry: ICacheEntry): string {
-	return `${entry.file.docId}_${entry.type}_${entry.key}`;
+	const version =
+		"fileVersion" in entry.file.resolvedUrl && entry.file.resolvedUrl.fileVersion !== undefined
+			? `_${entry.file.resolvedUrl.fileVersion}`
+			: "";
+	switch (entry.type) {
+		case snapshotWithLoadingGroupIdKey:
+		case snapshotKey: {
+			// example versioned entry: docId_4.0_snapshot_
+			// example non-versioned entry: docId_snapshot_
+			// The trailing '_' is included for consistency with existing cache entries
+			return `${entry.file.docId}${version}_${entry.type}_`;
```

Comment:
Maybe be better would be still end this key with entry.key as we should use key but then also set key to "" for now instead of fileVersion as we are getting from resolvedUrl. So, for existing keys it will still remain docId_snapshot_

Reply:
This is because as it seems weird to not use the key in creating the final key.

Reply:
i think that will also simplify the function, as there won't be a need for special casing based on type